HashRegistrar.sol

```diff
@@ -0,0 +1,282 @@
+// hash registrar using deposits
+
+contract Deed {
+    // The Deed is a contract intended simply to hold ether
+    // It can be controlled only by the registrar and can only send ether back to the owner
+    Registrar public registrar;
+    bytes32 public sealedBid;
+    bytes32 public hash;
+    address constant burn = 0xdead;
+    uint public creationDate;
+    address public owner;
+  
+    modifier noEther {
+        if (msg.value > 0) throw;
+        _
+    } 
+    
+    modifier onlyRegistrar {
+        if (msg.sender != address(registrar)) throw;
+        _
+    }
+    
+    function Deed(bytes32 _sealedBid, address _registrar) {
+        registrar = Registrar(_registrar);
+        sealedBid = _sealedBid;
+        creationDate = now;
+    }
+    
+    function unsealBid(bytes32 _hash, address _owner, uint _value, bytes32 _salt) noEther onlyRegistrar {
+        if (sha3(_hash, _owner, _value, _salt) != sealedBid) throw;
```

Comment:
Can this be modified to only run if _hash is 0? That way, it's guaranteed it can only run once, and therefore there's no way the original owner can steal it back by 'revealing' their bid again. I know this is `onlyRegistrar`, but the added self-contained security wouldn't hurt.